reflex-dom-core/src/Reflex/Dom/Main.hs

```diff
@@ -86,30 +87,31 @@ attachHydrationWidget
     -> IORef HydrationMode
     -> Maybe (IORef [(Node, HydrationRunnerT DomTimeline (DomCoreWidget ()) ())])
     -> EventChannel
-    -> PerformEventT DomTimeline DomHost (a, IORef (Maybe (EventTrigger DomTimeline ())))
+    -> PostBuildT DomTimeline (PerformEventT DomTimeline DomHost) a
      )
-  -> IO (a, FireCommand DomTimeline DomHost)
+  -> IO a
 attachHydrationWidget switchoverAction jsSing w = do
   hydrationMode <- liftIO $ newIORef HydrationMode_Hydrating
   rootNodesRef <- liftIO $ newIORef []
   events <- newChan
-  runDomHost $ flip runTriggerEventT events $ mdo
+  (result, _, _) <- runHostWithIO events runDomHost runDomHost $ flip runTriggerEventT events $ mdo
     (syncEvent, fireSync) <- newTriggerEvent
-    ((result, postBuildTriggerRef), fc@(FireCommand fire)) <- lift $ hostPerformEventT $ do
-      a <- w syncEvent hydrationMode (Just rootNodesRef) events
-      _ <- runWithReplace (return ()) $ delayedAction <$ syncEvent
-      pure a
-    mPostBuildTrigger <- readRef postBuildTriggerRef
-    lift $ forM_ mPostBuildTrigger $ \postBuildTrigger -> fire [postBuildTrigger :=> Identity ()] $ return ()
+
+    result <- lift $ w syncEvent hydrationMode (Just rootNodesRef) events
+    _ <- lift $ runWithReplace (return ()) $ delayedAction <$ syncEvent
+
     liftIO $ fireSync ()
     rootNodes <- liftIO $ readIORef rootNodesRef
-    let delayedAction = do
+    let delayedAction :: PostBuildT DomTimeline (PerformEventT DomTimeline DomHost) ()
+        delayedAction = do
           for_ (reverse rootNodes) $ \(rootNode, runner) -> do
             let hydrate = runHydrationRunnerT runner Nothing rootNode events
-            void $ runWithJSContextSingleton (runPostBuildT hydrate never) jsSing
+            -- Should this be lift, or distributeJSContextSingleton? Why do we use never?
```

Comment:
Could someone help me out with this?

Reply:
Hmm iirc the `PostBuildT` shouldn't really be here, but it was fairly awkward to get rid of it